internal/core/server/dockerstats.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+type DockerStats struct {
```

Comment:
Can we move this to the `providers/docker` directory and rename it to something like `ContainerStats`?